suitcase/Csv/__init__.py

```diff
@@ -0,0 +1,267 @@
+    directory : string, Path or Manager.
+        For basic uses, this should be the path to the output directory given
+        as a string or Path object. Use an empty string ``''`` to place files
+        in the current working directory.
+
+        In advanced applications, this may direct the serialized output to a
+        memory buffer, network socket, or other writable buffer. It should be
+        an instance of ``suitcase.utils.MemoryBufferManager`` and
+        ``suitcase.utils.MultiFileManager`` or any object implementing that
+        inferface. See the suitcase documentation (LINK ONCE WRITTEN) for
+        details.
+
+    file_prefix : str, optional
+        The first part of the filename of the generated output files. This
+        string may include templates as in ``{proposal_id}-{sample_name}-``,
+        which are populated from the RunStart document. The default value is
+        ``{uid}-`` which is guaranteed to be present and unique. A more
+        descriptive value depends on the application and is therefore left to
+        the user.
+
+    stack_images : Boolean
+        This indicates if we want one image per file (`stack_images` = `False`)
+        or many images per file (`stack_images` = `True`). If using
+        `stack_images` = `False` then an additional image number is added to
+        the file name.
+
+    **kwargs : kwargs
+        kwargs to be passed to ``pandas.Dataframe.to_csv``.
+    """
+    def __init__(self, directory, file_prefix='{uid}-', **kwargs):
+
+        if isinstance(directory, (str, Path)):
+            self.manager = suitcase.utils.MultiFileManager(directory)
+        else:
+            self.manager = directory
+
+        self.artifacts = self.manager._artifacts
+        self._streamnames = {}  # maps descriptor uids to stream_names
+        self._files = defaultdict(list)  # maps stream_name to file lists
+        self._filenames = {}  # map stream_name to file names of csv files
+        self._file_prefix = file_prefix
+        self._templated_file_prefix = ''
+        self._start_found = False
+
+        self._has_header = set()  # a set of uids to tell a file has a header
+
+        kwargs.setdefault('header', True)
+        self._initial_header_kwarg = kwargs['header']  # to set the headers
+        kwargs.setdefault('index_label', 'time')
+        kwargs.setdefault('mode', 'a')
+        self._kwargs = kwargs
+
+    def start(self, doc):
+        '''Extracts `start` document information for formatting file_prefix.
+
+        This method checks that only one `start` document is seen and formats
+        `file_prefix` based on the contents of the `start` document.
+
+        Parameters:
+        -----------
+        doc : dict
+            RunStart document
+        '''
+
+        # raise an error if this is the second `start` document seen.
+        if self._start_found:
+            raise RuntimeError(
+                "The serializer in suitcase.tiff expects documents from one "
```

Comment:
tiff -> csv

Reply:
or `suitcase.Csv`? I saw it was renamed on top of this PR...

Reply:
done